### PR TITLE
958 enrollment card error

### LINKED ISSUE #958: Enrollment/Funding validation errors should show in the card as an exclamation point
- When I have the "Last reporting period must be after first" error, 
Screen Shot 2020-12-29 at 11 14 19 AM

- The enrollment card should display with an exclamation point after the entered value. 
Screen Shot 2020-12-29 at 11 18 47 AM

### PR DESCRIPTION
**Pull request by mmazanec22**
Closes #958

### CODE REVIEW

File: client/src/utils/fundingBeginsAfterEnrollmentEntry.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import { ValidationError } from 'class-validator';
+import { Funding } from '../shared/models';
+
+export function fundingBeginsAfterEnrollmentEntry(funding: Funding): boolean {
+  const firstReportingPeriodErrors = funding.validationErrors?.filter(
```
Comment (JordanGuinn) [inline client/src/utils/fundingBeginsAfterEnrollmentEntry.ts:5]: Super nitpick and really moreso just a personal preference thing and _by all means feel free to ignore me_, BUT you could easily cut this down to:

```
return funding.validationErrors?.some(
   (e) => e.property === 'firstReportingPeriod'
);
```

if you wanted.  Aaaaaaand I'm done 😅😁

**Review by JordanGuinn — APPROVED** [review]

File: client/src/components/Forms/FamilyIncome/Form.tsx [hunk 2]
```diff
@@ -52,9 +52,9 @@ export const doesFamilyIncomeFormHaveErrors = (
   const incomeDeterminationsHaveError = child?.family?.incomeDeterminations
     ?.length
     ? !!getValidationStatusForFields(
-      child.family.incomeDeterminations,
-      incomeDeterminationFields
-    )
+        child.family.incomeDeterminations,
+        incomeDeterminationFields
+      )
```
Comment (bamader) [inline client/src/components/Forms/FamilyIncome/Form.tsx:57]: I feel like I've seen this specific spacing change in like 4 separate PRs 🤣

**Review by bamader — APPROVED** [review]

File: client/src/utils/fundingBeginsAfterEnrollmentEntry.ts [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import { Funding } from '../shared/models';
+
+export function fundingBeginsAfterEnrollmentEntry(
```
Comment (jlhogan) [inline client/src/utils/fundingBeginsAfterEnrollmentEntry.ts:3]: consider naming this `fundingHasFirstReportingPeriodError` or something that belies the general nature of the check

Comment (jlhogan) [reply client/src/utils/fundingBeginsAfterEnrollmentEntry.ts:3]: or even more general, a function that takes an array of validation errors and a field name and returns if an error exists. Then we can use that for all fields displayed in cards and solve this problem writ large

**Pull request merged**